packages/core/src/components/Tabs/Tabs.jsx

```diff
@@ -134,6 +175,75 @@ export class Tabs extends React.PureComponent {
     }
   }
 
+  /**
+   * Build tabs array to handle arrow keypress events accurately
+   */
+  tabsArrayLen() {
+    return React.Children.toArray(this.props.children)
+      .filter(isTab);
+  }
+
+  /**
+   * Find the current tab index on left or right arrow keypress
+   * @param {Array} React Tab components
+   * @param {String} ID prop of currently selected Tab
+   * @return {Number} Returns index of currently selected array item
+   */
+  tabFindIndex(tabsArr, findId) {
+    const index = tabsArr.findIndex(elem => elem.props.id === findId);
```

Comment:
Could you use `this.tabs` here rather than passing it in as an argument?

Reply:
Nit: Still curious about this. Looks like `tabFindIndex` is only ever called with `this.panelChildren()` as the first argument. We could dry up `tabSwitchLeft` and `tabSwitchRight` by moving that logic here and getting rid of the `tabsArr` argument.